File: frontend-angular/src/app/services/user-address.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { UserAddressResponseDTO } from '../interface/user-address-response-dto';
import { UserAddressRequestDTO } from '../interface/user-address-request-dto';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class UserAddressService {

  private readonly BASE_URL = 'http://localhost:8200/api/v1/user/addresses';

  constructor(private http: HttpClient) {}

  getAddresses(userId: string): Observable<UserAddressResponseDTO[]> {
    return this.http.get<UserAddressResponseDTO[]>(`${this.BASE_URL}/${userId}`);
  }

  addAddress(userId: string, address: UserAddressRequestDTO): Observable<UserAddressResponseDTO> {
    return this.http.post<UserAddressResponseDTO>(`${this.BASE_URL}/${userId}`, address);
  }

  deleteAddress(addressId: number): Observable<void> {
    return this.http.delete<void>(`${this.BASE_URL}/${addressId}`);
  }
}

File: backend/src/main/java/com/elhabhab/backend/service/user/address/UserAddressService.java
package com.elhabhab.backend.service.user.address;


import com.elhabhab.backend.dto.request.UserAddressRequestDTO;
import com.elhabhab.backend.dto.response.UserAddressResponseDTO;

import java.util.List;
import java.util.UUID;

public interface UserAddressService {


    public List<UserAddressResponseDTO> getAddresses(UUID userId);

    public UserAddressResponseDTO addAddress(UUID userId, UserAddressRequestDTO dto) ;

    public void deleteAddress(Long addressId) ;
}
